app/renderer/js/tray.ts

```diff
@@ -183,6 +185,34 @@ ipcRenderer.on('tray', (_event: Event, arg: number): void => {
 	}
 });
 
+const monitorTray = (): void => {
+	if (process.platform !== 'linux') {
+		return;
+	}
+
+	/* Here we check if the screen has been locked and unlocked.
+		If the monitor returns SCREEN_UNLOCKED, then that has to be our case
+		And we destroy and recreate the tray
+	*/
+	const monitor = exec(`
+	  dbus-monitor --session "type='signal',interface='org.gnome.ScreenSaver'" |
+	  while read x; do
+		case "$x" in 
+		  *"boolean true"*) echo SCREEN_LOCKED;;
+		  *"boolean false"*) echo SCREEN_UNLOCKED;;  
+		esac
+	  done
+	`);
```

Comment:
* We should write the logic in JS instead of copying weird shell script snippets from StackOverflow.
* Instead of looking for `org.gnome.ScreenSaver` signals, I think we want to look for `org.freedesktop.DBus.NameAcquired("org.kde.StatusNotifierWatcher")`. (AppIndicator is based on KStatusNotifierItem.)
* Instead of forking a `dbus-monitor`, can we use a JS D-Bus library?